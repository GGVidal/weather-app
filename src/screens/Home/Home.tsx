import React, { FC } from "react";
import { Text } from "../../components/Text";
import WeatherIcon from "../../assets/WeatherIcon.svg";
import {
  Container,
  ImageContainer,
  TextContainer,
  SubtitleContainer,
} from "./style";

export const Home: FC = () => {
  return (
    <Container>
      <ImageContainer>
        <WeatherIcon width={150} height={150} />
      </ImageContainer>
      <TextContainer>
        <Text
          fontSize="28px"
          lineHeight="32px"
          color="#313341"
          fontWeight="500"
        >
          Descubra o clima na sua cidade
        </Text>
        <SubtitleContainer>
          <Text
            fontSize="16px"
            lineHeight="20px"
            color="#9A938C"
            fontWeight="400"
          >
            Veja sua localização no mapa e o clima na região
          </Text>
        </SubtitleContainer>
      </TextContainer>
    </Container>
  );
};
