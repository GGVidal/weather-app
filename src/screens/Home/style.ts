import styled from "styled-components/native";

export const Container = styled.View`
  padding-horizontal: 36px;
`;

export const ImageContainer = styled.View`
  align-items: center;
  justify-content: center;
  margin-top: 100px;
`;

export const TextContainer = styled.View`
  margin-top: 50px;
  align-items: center;
`;

export const SubtitleContainer = styled.View`
  margin-top: 6px;
  align-items: center;
`;

export const BeginButtonContainer = styled.View`
  margin-top: 180px;
`;
