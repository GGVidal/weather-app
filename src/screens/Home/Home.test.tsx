import React from "react";
import TestRenderer from "react-test-renderer";
import { render, fireEvent, act } from "@testing-library/react-native";
import "@testing-library/jest-dom";

import { Home } from "./Home";

jest.mock("expo-location", () => ({
  requestForegroundPermissionsAsync: () => jest.fn(),
}));

jest.mock("expo-linear-gradient", () => ({
  LinearGradient: () => {},
}));

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});
describe("renders correctly", () => {
  it("Render home", () => {
    const renderer = render(<Home />);
    expect(renderer).toMatchSnapshot();
  });

  // it("Click iniciar", async () => {
  //   const { getByTestId } = render(<Home />);
  //   const beginButton = getByTestId("button-permission");
  //   fireEvent.press(beginButton);
  // });
});
