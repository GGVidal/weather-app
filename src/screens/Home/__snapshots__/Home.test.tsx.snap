// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Render home 1`] = `null`;
